tests/expected-after-fixing/lint-with-problems.md

```diff
@@ -0,0 +1,45 @@
+# This file has something to lint
+
+```php
+class MyClass
+{
+    private string $myProperty = 'this is the value';
+}
+```
+
+There's plenty of problems with that one.
+
+```php
+namespace App;
+
+class MyClass
+{
+    private string $myProperty = 'this is the value';
+}
+```
+
+There should be no linting issues with that block.
+
+```js
+// This one's just here so that we can check the num counter works correctly.
+// This block should be skipped entirely.
+```
+
+```PHP
+class AnotherClass
+{
+    private string $anotherProperty = 'this is the value';
+}
+```
+
+And then that one should have linting failures again - note the capital PHP, which should still be picked up.
+
+For some reason this one doesn't get auto-fixed.
+
+- this one will be indented
+  ```php
+  class FinalClass {
+      private string $lastProperty='this is the value';
+  }
+  ```
+- And it'll also have linting failures
```

Comment:
The note about this one not getting auto-fixed is obviously pointing out a bug.

Once this PR is merged I'll raise a new issue about it, but I couldn't see an obvious way to resolve it. I figured it was better to have this working in the 90% of scenarios that it needs to for now, and we can fix that later on.